api/src/main/java/jakarta/data/Operator.java

```diff
@@ -24,5 +24,40 @@ public enum Operator {
     IN,
     LESS_THAN,
     LESS_THAN_EQUAL,
-    LIKE
+    LIKE,
+    NOT_EQUAL,
+    NOT_IN,
+    NOT_LIKE;
+
+    /**
+     * Returns the operator that is the negation of this operator.
+     *
+     * @return the operator that is the negation of this operator.
+     */
+    Operator negate() {
+        switch (this) {
+            case EQUAL:
+                return NOT_EQUAL;
+            case GREATER_THAN:
+                return LESS_THAN_EQUAL;
+            case GREATER_THAN_EQUAL:
+                return LESS_THAN;
+            case IN:
+                return NOT_IN;
+            case LESS_THAN:
+                return GREATER_THAN_EQUAL;
+            case LESS_THAN_EQUAL:
+                return GREATER_THAN;
+            case LIKE:
+                return NOT_LIKE;
+            case NOT_EQUAL:
+                return EQUAL;
+            case NOT_IN:
+                return IN;
+            case NOT_LIKE:
+                return LIKE;
+            default: // should be unreachable
```

Comment:
Should it be the opposite?
I mean, NOT_EQUAL returning EQUAL

Reply:
> Should it be the opposite? I mean, NOT_EQUAL returning EQUAL

Otavio, I don't understand your comment in relation to the code.  The negate method code switches NOT_EQUAL to EQUAL, and it switches EQUAL to NOT_EQUAL,

```
            case EQUAL:
                return NOT_EQUAL;
            case GREATER_THAN:
                return LESS_THAN_EQUAL;
            case GREATER_THAN_EQUAL:
                return LESS_THAN;
            case IN:
                return NOT_IN;
            case LESS_THAN:
                return GREATER_THAN_EQUAL;
            case LESS_THAN_EQUAL:
                return GREATER_THAN;
            case LIKE:
                return NOT_LIKE;
            case NOT_EQUAL:
                return EQUAL;
            case NOT_IN:
                return IN;
            case NOT_LIKE:
                return LIKE;
            default: // should be unreachable
```

Maybe I am overlooking something. Could you add a suggestion with the specific correction you want to see made?

Reply:
That was my point, so, please ignore it.